frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1530,29 +1533,48 @@ impl<T: Config> Pallet<T> {
 		// - signed phase was complete or not started, in which case finalization is idempotent and
 		//   inexpensive (1 read of an empty vector).
 		let _ = Self::finalize_signed_phase();
-		<QueuedSolution<T>>::take()
-			.map_or_else(
-				|| {
-					T::Fallback::elect()
-						.map_err(|fe| ElectionError::Fallback(fe))
-						.map(|supports| (supports, ElectionCompute::Fallback))
-				},
-				|ReadySolution { supports, compute, .. }| Ok((supports, compute)),
-			)
-			.map(|(supports, compute)| {
-				Self::deposit_event(Event::ElectionFinalized { election_compute: Some(compute) });
-				if Self::round() != 1 {
-					log!(info, "Finalized election round with compute {:?}.", compute);
-				}
-				supports
-			})
-			.map_err(|err| {
-				Self::deposit_event(Event::ElectionFinalized { election_compute: None });
+		match <QueuedSolution<T>>::take() {
```

Comment:
Sorry but this block of code is exactly why we have `or_else` and alike, and you have pretty much undid it :( 

The discussion in https://github.com/paritytech/substrate/pull/12069#discussion_r953381297 didn't ask for remove all the map functions. It just mentioned that you shouldn't map from `ReadySolution { x y z } to (x, y, z)`. That should just be an `or_else`, not `map_or_else`.

Reply:
I understood the discussion before , but i didn't thought of using `ok_or()` first, so ... :(